File: LiveDataSampleJava/app/src/main/java/com/websarva/wings/android/livedatasamplejava/data/local/CocktailmemoDAO.java
package com.websarva.wings.android.livedatasamplejava.data.local;

import com.google.common.util.concurrent.ListenableFuture;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface CocktailmemoDAO {
	@Query("SELECT * FROM cocktailmemos WHERE id = :id")
	ListenableFuture<Cocktailmemo> findByPK(int id);

	@Insert
	ListenableFuture<Long> insert(Cocktailmemo cocktailmemo);

	@Delete
	ListenableFuture<Integer> delete(Cocktailmemo cocktailmemo);
}

File: LiveDataSampleKotlin/app/src/main/java/com/websarva/wings/android/livedatasamplekotlin/data/local/CocktailmemoDAO.kt
package com.websarva.wings.android.repositorysamplekotlin.data.local

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface CocktailmemoDAO {
	@Query("SELECT * FROM cocktailmemos WHERE id = :id")
	suspend fun findByPK(id: Int): Cocktailmemo?

	@Insert
	suspend fun insert(cocktailmemo: Cocktailmemo): Long

	@Delete
	suspend fun delete(cocktailmemo: Cocktailmemo): Int
}
